src/data/polynomial/erase_lead.lean

```diff
@@ -0,0 +1,150 @@
+-/
+
+noncomputable theory
+open_locale classical
+
+open polynomial finsupp finset
+
+namespace polynomial
+
+variables {R : Type*} [semiring R] {f : polynomial R}
+
+/-- `erase_lead f` for a polynomial `f` is the polynomial obtained by
+subtracting from `f` the leading term of `f`. -/
+def erase_lead (f : polynomial R) : polynomial R :=
+finsupp.erase f.nat_degree f
+
+section erase_lead
+
+lemma erase_lead_support (f : polynomial R) :
+  f.erase_lead.support = f.support.erase f.nat_degree :=
+by convert rfl
+
+lemma erase_lead_coeff (i : ℕ) :
+  f.erase_lead.coeff i = if i = f.nat_degree then 0 else f.coeff i :=
+by convert rfl
+
+@[simp] lemma erase_lead_coeff_nat_degree : f.erase_lead.coeff f.nat_degree = 0 :=
+finsupp.erase_same
+
+lemma erase_lead_coeff_of_ne (i : ℕ) (hi : i ≠ f.nat_degree) :
+  f.erase_lead.coeff i = f.coeff i :=
+finsupp.erase_ne hi
+
+@[simp] lemma erase_lead_zero : erase_lead (0 : polynomial R) = 0 :=
+finsupp.erase_zero _
+
+@[simp] lemma erase_lead_add_monomial_nat_degree_leading_coeff (f : polynomial R) :
+  f.erase_lead + monomial f.nat_degree f.leading_coeff = f :=
+begin
+  ext i,
+  simp only [erase_lead_coeff, coeff_monomial, coeff_add, @eq_comm _ _ i],
+  split_ifs with h,
+  { subst i, simp only [leading_coeff, zero_add] },
+  { exact add_zero _ }
+end
+
+@[simp] lemma erase_lead_add_C_mul_X_pow (f : polynomial R) :
+  f.erase_lead + (C f.leading_coeff) * X ^ f.nat_degree = f :=
+by rw [C_mul_X_pow_eq_monomial, erase_lead_add_monomial_nat_degree_leading_coeff]
+
+@[simp] lemma self_sub_monomial_nat_degree_leading_coeff {R : Type*} [ring R] (f : polynomial R) :
+  f - monomial f.nat_degree f.leading_coeff = f.erase_lead :=
+(eq_sub_iff_add_eq.mpr (erase_lead_add_monomial_nat_degree_leading_coeff f)).symm
+
+@[simp] lemma self_sub_C_mul_X_pow {R : Type*} [ring R] (f : polynomial R) :
+  f - (C f.leading_coeff) * X ^ f.nat_degree = f.erase_lead :=
+by rw [C_mul_X_pow_eq_monomial, self_sub_monomial_nat_degree_leading_coeff]
+
+lemma erase_lead_ne_zero (f0 : 2 ≤ f.support.card) : erase_lead f ≠ 0 :=
+begin
+  have fn0 : f ≠ 0,
+  { rintro rfl, simpa only [card_empty, le_zero_iff_eq, support_zero, two_ne_zero] using f0 },
+  rw [ne.def, ← support_eq_empty, erase_lead_support],
+  apply @ne_empty_of_mem _ (nat_trailing_degree f),
+  apply mem_erase_of_ne_of_mem _ (nat_trailing_degree_mem_support_of_nonzero fn0),
+  rw [nat_degree_eq_support_max' fn0, nat_trailing_degree_eq_support_min' fn0],
+  exact ne_of_lt (finset.min'_lt_max'_of_card _ f0)
+end
+
+@[simp] lemma nat_degree_not_mem_erase_lead_support : f.nat_degree ∉ (erase_lead f).support :=
+by convert not_mem_erase _ _
+
+lemma ne_nat_degree_of_mem_erase_lead_support {a : ℕ} (h : a ∈ (erase_lead f).support) :
+  a ≠ f.nat_degree :=
+by { rintro rfl, exact nat_degree_not_mem_erase_lead_support h }
+
+lemma erase_lead_nat_degree_lt (f0 : 2 ≤ f.support.card) :
+  (erase_lead f).nat_degree < f.nat_degree :=
```

Comment:
You prove `≤` version later in this file. If you move this lemma below, then you'll only have to prove `≠`. This will make the proof 3 lines long.

Reply:
I tried, but the lemma below is stated for ```degree```, not ```nat_degree```.  This means that my proof of this fact would actually be longer than the given one, since I am not very versed in working with ```option```s...

Reply:
You had a `nat_degree` version too. Golfed.